### DISCUSSION TITLE
Unable to create process at all with virtual environment binary

### DISCUSSION THREAD

**Discussion by WizzyGeek** [pypa/virtualenv#2244] [Q&A] [answered]
# Context

Python Install - `c:\users\username withspace\appdata\local\programs\python\python39\python.exe`
Project Location - `C:\.dev\project`

Project Directory Tree
```
├───.venv
│   └───[Standard virtualenv tree]
├───package
│   ├───__init__.py
│   └───__main__.py
├───README.md
├───pyproject.toml
└───setup.cfg
```

Command used to generate venv - `virtualenv .venv/`
OS - `Windows 11 21H2 Build 22000.346`
Python - `Python 3.9.4 (tags/v3.9.4:1f2e308, Apr  6 2021, 13:40:21) [MSC v.1928 64 bit (AMD64)] on win32`

# Issue

So I created a virtual env on windows 11
with `virtualenv .venv/`

```
C:\.dev\trainer>virtualenv .venv/ 
created virtual environment CPython3.9.4.final.0-64 in 1402ms
  creator CPython3Windows(dest=C:\.dev\trainer\.venv, clear=False, no_vcs_ignore=False, global=False)
  seeder FromAppData(download=False, pip=bundle, setuptools=bundle, wheel=bundle, via=copy, app_data_dir=C:\Users\username withspace\AppData\Local\pypa\virtualenv)
    added seed packages: pip==21.3.1, setuptools==58.3.0, wheel==0.37.0
  activators BashActivator,BatchActivator,FishActivator,NushellActivator,PowerShellActivator,PythonActivator
```

Then I proceeded to activate and use the virtual environment

```
PS C:\.dev\trainer> & c:/.dev/trainer/.venv/Scripts/Activate.ps1
(.venv) PS C:\.dev\trainer> python
Unable to create process using 'c:\users\username withspace\appdata\local\programs\python\python39\python.exe'
(.venv) PS C:\.dev\trainer> pip
Unable to create process using 'C:\Users\username withspace\AppData\Local\Programs\Python\Python39\python.exe "C:\.dev\trainer\.venv\Scripts\pip.exe" '
```

The launcher **completely fails** to launch python

## Things Tried
- `py -3.9 -m pip install distlib -U`
- `py -3.9 -m pip install pip -U`
- `py -3.9 -m pip install virtualenv -U`
- `py -3.9 -m venv .venv/` (over the existing virtualenv created by `virtualenv .venv/`)
  ```
  (.venv) PS C:\.dev\trainer> py -3.9 -m venv .venv/
  Error: Command '['C:\\.dev\\trainer\\.venv\\Scripts\\python.exe', '-Im', 'ensurepip', '--upgrade', '--default-pip']' returned non-zero exit status 101.
  ```
- `py -3.9 -m venv .venv/ --without-pip`
  Passes but the problem still persists
  ```
  (.venv) PS C:\.dev\trainer> py -3.9 -m venv .venv/ --without-pip
  (.venv) PS C:\.dev\trainer> python
  Unable to create process using 'C:\Users\username withspace\AppData\Local\Programs\Python\Python39\python.exe'
  ```

<hr>

I have been juggling between various issues until I decided to ask a question here.
Does anyone have any idea what is causing this issue? and how can I fix it?

**Comment by WizzyGeek** [comment]
The launcher works fine when run as admin, but it's an inconvenience to launch the terminal as admin everytime

**Comment by WizzyGeek** [comment, marked as answer]
Linking a stack overflow answer for anyone who stumbles upon this in the future

https://stackoverflow.com/questions/61641280/python3-8-venv-returned-exit-status-101/62207756#62207756

**Comment by gaborbernat** [comment]
This discussion was self-answered with a solution. The issue was caused by spaces in the Windows user path, and the Stack Overflow answer you linked provides the fix: https://stackoverflow.com/questions/61641280/python3-8-venv-returned-exit-status-101/62207756#62207756. Closing as answered.
